clang/lib/CIR/CodeGen/CIRGenFunction.cpp

```diff
@@ -757,6 +757,20 @@ mlir::Value CIRGenFunction::createLoad(const VarDecl *VD, const char *Name) {
                                 addr.getElementType(), addr.getPointer());
 }
 
+void CIRGenFunction::emitCXXGuardedInitBranch(mlir::Value needsInit,
+                                               mlir::Block *initBlock,
+                                               mlir::Block *noInitBlock,
+                                               GuardKind kind,
+                                               const VarDecl *varDecl) {
+  assert((kind == GuardKind::tlsGuard || varDecl) && "no guarded variable");
+
+  if (MissingFeatures::metaDataNode())
+    llvm_unreachable("NYI");
+
+  builder.createCondBr(getLoc(varDecl->getLocation()), needsInit, initBlock,
```

Comment:
This should be a cir::IfOp

Reply:
Yup, that's TODO, didn't want to spend time on that on the draft pass before you gave high level feedback.